### PR TITLE
Non lint highlighting

### PR DESCRIPTION
**Pull request by schlank**
Uses Text Highlighting and Hints instead of using the default error and warning annotations.  Developers requested that it not interfere with lint warning and keep files from being "green."

### CODE REVIEW

**Comment by schlank** [comment]
Next step will be to give the user the option to use Hexar warnings with or without lint by adding persisted configuration.

File: src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java [hunk 1]
```diff
@@ -17,19 +33,79 @@ public void annotate(@NotNull PsiElement psiElement, @NotNull AnnotationHolder a
...
         ModelParser modelParser = new ModelParser();
         for (ModelError error : modelParser.parse(psiElement)) {
-            annotationHolder.createWarningAnnotation(error.getElementToTag(), error.messageForError());
+            annotateElement(error.getElementToTag(), error.messageForError());
         }
 
         lastElement = psiElement;
     }
+
+    private void annotateElement(final PsiElement psiElement, final String message) {
```
Comment (schlank) [inline src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java]: Looks like I needed RoboHexar to remind me to add javadocs.

File: src/com/smilingrob/plugin/robohexar/JavaDocError.java [hunk 2]
```diff
@@ -16,10 +16,6 @@
         MISSING_JAVA_DOC
     }
 
-    /**
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/JavaDocError.java:19]: Removing docs?

File: src/com/smilingrob/plugin/robohexar/JavaDocParser.java [hunk 3]
```diff
@@ -80,13 +80,12 @@ private boolean hasTestAnnotation(@NotNull PsiMethodImpl psiMethod) {
...
-     * Determines whether or not the given method is a getter or a setter.
-     *
-     * @param psiMethodImpl method node.
-     * @return true if the given method is a getter or a setter.
-     */
+    * Determines whether or not the given method is a getter or a setter.
+    *
+    * @param psiMethodImpl method node.
+    * @return true if the given method is a getter or a setter.
+    */
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/JavaDocParser.java:88]: What is up with these format changes?

File: src/com/smilingrob/plugin/robohexar/ModelError.java [hunk 4]
```diff
@@ -42,11 +42,11 @@ public PsiElement getElementToTag() {
     public String messageForError() {
         switch (mType) {
             case ACCESSORS_NOT_ALPHABETIZED:
-                return "Accessors not alphabetized in class.";
+                return "Hey dummy, accessors are not alphabetized in this class.";
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/ModelError.java]: Wow it's fully accusatory. I don't think I've even seen Hexar say stuff like that, even jokingly.

File: src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java [hunk 5]
```diff
@@ -17,19 +33,79 @@ public void annotate(@NotNull PsiElement psiElement, @NotNull AnnotationHolder a
 
         GrammarParser grammarParser = new GrammarParser();
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java:48]: These weak warnings look good.

File: src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java [hunk 6]
```diff
@@ -17,19 +33,79 @@ public void annotate(@NotNull PsiElement psiElement, @NotNull AnnotationHolder a
...
         lastElement = psiElement;
     }
+
+    private void annotateElement(final PsiElement psiElement, final String message) {
+
+        final TextAttributes textattributes = new TextAttributes(null, null, JBColor.RED, EffectType.WAVE_UNDERSCORE, Font.ITALIC);
+
+        ApplicationManager.getApplication().invokeLater(new Runnable() {
+            public void run() {
+                final Project project = psiElement.getProject();
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java]: The contents of this runnable could be a function.

File: src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java [hunk 7]
```diff
@@ -17,19 +33,79 @@ public void annotate(@NotNull PsiElement psiElement, @NotNull AnnotationHolder a
...
+                final Editor editor = editorManager.getSelectedTextEditor();
+                if (editor != null) {
+                    editor.getMarkupModel().addRangeHighlighter(
+                            psiElement.getTextOffset(),
+                            psiElement.getTextOffset() + psiElement.getTextLength(),
+                            HighlighterLayer.WARNING,
+                            textattributes,
+                            HighlighterTargetArea.EXACT_RANGE);
+
+                    editor.addEditorMouseListener(new EditorMouseListener() {
```
Comment (robert-wallis) [inline src/com/smilingrob/plugin/robohexar/RoboHexarAnnotator.java]: This could be pulled out to an inner class.

**Comment by robert-wallis** [comment]
Looks good.  Feel free to merge, ship, and update the plugin, anytime you want.  I don't mind doing PRs if you want.

After watching https://cleancoders.com/episode/clean-code-episode-5/show I agree that JavaDocs everywhere are not necessary.  And that comments should be in code to explain weird things, and that indicates a failure to write readable code.  So I am not going to use this plugin anymore.

File: src/com/smilingrob/plugin/robohexar/ModelError.java (hunk 4 above)
Comment (schlank) [inline src/com/smilingrob/plugin/robohexar/ModelError.java]: k, Hexar agrees.  Hexar and Robert have not sense of humor. :)

File: src/com/smilingrob/plugin/robohexar/JavaDocParser.java (hunk 3 above)
Comment (schlank) [inline src/com/smilingrob/plugin/robohexar/JavaDocParser.java:88]: I removed the javadoc to set off RoboHexar for testing.  I put them back and they were off.  I used github directly.

**Pull request merged**